### PR TITLE
Fixes for localized strings in template tags

### PR DESCRIPTION
**Pull request by frankiejarrett**
Adding context where appropriate and removing screen reader HTML from localized string.

### CODE REVIEW

File: inc/template-tags.php [hunk 1]
```diff
@@ -76,7 +76,12 @@ function twentysixteen_entry_meta() {
 
 	if ( ! is_singular() && ! post_password_required() && ( comments_open() || get_comments_number() ) ) {
 		echo '<span class="comments-link">';
-		comments_popup_link( sprintf( __( 'Leave a comment<span class="screen-reader-text"> on %s</span>', 'twentysixteen' ), get_the_title() ) );
+		comments_popup_link(
+			sprintf(
+				_x( 'Leave a comment %s', 'Screen reader text', 'twentysixteen' ),
+				'<span class="screen-reader-text">' . sprintf( esc_html_x( 'on %s', 'Name of current post', 'twentysixteen' ), esc_html( get_the_title() ) ) . '</span>'
```
Comment (grappler) [inline inc/template-tags.php]: The phrase cannot be split as other languages have a different sentence structure. The word "on" may come in the beginning of the phrase.

File: inc/template-tags.php [hunk 2]
```diff
@@ -181,9 +186,8 @@ function twentysixteen_post_thumbnail() {
 function twentysixteen_excerpt_more( $more ) {
 	$link = sprintf( '<a href="%1$s" class="more-link">%2$s</a>',
 		esc_url( get_permalink( get_the_ID() ) ),
-		/* translators: %s: Name of current post */
-		sprintf( __( 'Continue reading %s', 'twentysixteen' ), '<span class="screen-reader-text">' . esc_html( get_the_title( get_the_ID() ) ) . '</span>' )
-		);
+		sprintf( _x( 'Continue reading %s', 'Name of current post', 'twentysixteen' ), '<span class="screen-reader-text">' . esc_html( get_the_title( get_the_ID() ) ) . '</span>' )
```
Comment (grappler) [inline inc/template-tags.php:184]: The way with the translation comment works fine. I don't think we need to change it to `_x()`

Comment (frankiejarrett) [inline inc/template-tags.php:184]: Will an inline comment even be seen by translators in Poedit or Glotpress? AFAIK contextual hints for the translator belong inside `_x` for this reason, and inline PHP comments should not be used for this. Please correct me if I'm wrong :smile:

File: inc/template-tags.php (hunk 1 above)
Comment (frankiejarrett) [inline inc/template-tags.php]: Makes sense. Still a shame to have HTML in translation string. I'll ponder this for a while and maybe will have something clever to show later :smile:

File: inc/template-tags.php (hunk 2 above)
Comment (grappler) [inline inc/template-tags.php:184]: I have not used Poedit recently but the developer told me he going to add support for it. GlotPress has always supported it. You can see it showing up in the German translation: https://translate.wordpress.org/projects/wp-themes/twentysixteen/de/default?filters%5Bterm%5D=Continue+reading+%25s&filters%5Buser_login%5D=&filters%5Bstatus%5D=current_or_waiting_or_fuzzy_or_untranslated&filter=Filter&sort%5Bby%5D=priority&sort%5Bhow%5D=desc

**Comment by karmatosed** [comment]
Can you refresh this to avoid conflicts please?

File: inc/template-tags.php (hunk 1 above)
Comment (frankiejarrett) [inline inc/template-tags.php]: Reverted back in https://github.com/fjarrett/twentysixteen/commit/f24b65536f695d1350ff938cb7ea27fbd1dffacf

**Comment by frankiejarrett** [comment]
@karmatosed Done.

**Comment by valendesigns** [comment]
Needs a refresh and build is failing but Travis is not being helpful about why.

**Comment by karmatosed** [comment]
Noting this is flagged for discussion in the next meeting: Friday 16:00 UTC in #core-themes.

**Comment by karmatosed** [comment]
Can we get a refreshed PR and then I'll merge this @fjarrett.

**Comment by frankiejarrett** [comment]
@karmatosed Done.

**Comment by karmatosed** [comment]
Awesome - thanks this is now merged!

**Pull request merged**